repos/omeka-i18n-module/src/Mvc/Router/SkippableSegment.php

```diff
@@ -0,0 +1,130 @@
+        }
+        if (!isset($options['skippable'])) {
+            $options['skippable'] = array();
+        }
+        return new static($options['route'], $options['constraints'], $options['defaults'], $options['skippable']);
+    }
+    /**
+     * Build a path.
+     *
+     * @param array $parts
+     * @param array $mergedParams
+     * @param bool $isOptional
+     * @param bool $hasChild
+     * @param array $options
+     * @return string
+     * @throws Exception\InvalidArgumentException
+     * @throws Exception\RuntimeException
+     */
+    protected function buildPath(array $parts, array $mergedParams, $isOptional, $hasChild, array $options)
+    {
+        if ($this->translationKeys) {
+            if (!isset($options['translator']) || !$options['translator'] instanceof Translator) {
+                throw new Exception\RuntimeException('No translator provided');
+            }
+            $translator = $options['translator'];
+            $textDomain = (isset($options['text_domain']) ? $options['text_domain'] : 'default');
+            $locale = (isset($options['locale']) ? $options['locale'] : null);
+        }
+        $path = '';
+        $skip = true;
+        $skippable = false;
+        foreach ($parts as $part) {
+            switch ($part[0]) {
+                case 'literal':
+                    $path .= $part[1];
+                    break;
+                case 'parameter':
+                    $skippable = true;
+                    if (!empty($this->skippable[$part[1]]) && (!isset($mergedParams[$part[1]]) ||
+                            array_key_exists($part[1], $this->defaults) && $mergedParams[$part[1]] === $this->defaults[$part[1]])) {
+                        $this->assembledParams[] = $part[1];
+                        break;
+                    } elseif (!isset($mergedParams[$part[1]])) {
+                        if (!$isOptional || $hasChild) {
+                            throw new Exception\InvalidArgumentException(sprintf('Missing parameter "%s"', $part[1]));
+                        }
+                        return '';
+                    } elseif (!$isOptional || $hasChild || !isset($this->defaults[$part[1]]) || $this->defaults[$part[1]] !== $mergedParams[$part[1]]) {
+                        $skip = false;
+                    }
+                    $path .= $this->encode($mergedParams[$part[1]]);
+                    $this->assembledParams[] = $part[1];
+                    break;
+                case 'optional':
+                    $skippable = true;
+                    $optionalPart = $this->buildPath($part[1], $mergedParams, true, $hasChild, $options);
+                    if ($optionalPart !== '') {
+                        $path .= $optionalPart;
+                        $skip = false;
+                    }
+                    break;
+                case 'translated-literal':
+                    $path .= $translator->translate($part[1], $textDomain, $locale);
```

Comment:
Are we using this? Could be potentially useful.

Reply:
No currently, hadn't noticed the extra type